pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -188,6 +212,80 @@ def _primitive_bind_call(cls, mps, wires, id=None):
             return type.__call__(cls, mps=mps, wires=wires, id=id)  # pragma: no cover
         return cls._primitive.bind(*mps, wires=wires, id=id)
 
+    def decomposition(self):  # pylint: disable=arguments-differ
+        filtered_hyperparameters = {
+            key: value for key, value in self.hyperparameters.items() if key != "input_wires"
+        }
+        return self.compute_decomposition(
+            self.parameters, wires=self.hyperparameters["input_wires"], **filtered_hyperparameters
+        )
+
```

Comment:
Can we store `self.hyperparameters["input_wires"]` in a variable, then simply `del self.hyperparameters["input_wires"]`, and do `**self.hyperparameters`?

Reply:
I have just tested it. This deletes it from the hyperparameters permanently (and not only inside the function).
This is one way to make the copy so I'd better leave it like this 😄